unity-client/Assets/Scripts/MainScripts/DCL/Controllers/QuestsController/QuestsController.cs

```diff
@@ -0,0 +1,174 @@
+﻿using DCL.Helpers;
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using UnityEngine;
+using Random = UnityEngine.Random;
+
+namespace DCL.QuestsController
+{
+    public delegate void QuestProgressed(string questId);
+    public delegate void QuestCompleted(string questId);
+    public delegate void SectionCompleted(string questId, string sectionId);
+    public delegate void SectionUnlocked(string questId, string sectionId);
+    public delegate void TaskProgressed(string questId, string sectionId, string taskId);
+
+    public interface IQuestsController
+    {
+        event QuestProgressed OnQuestProgressed;
+        event QuestCompleted OnQuestCompleted;
+        event SectionCompleted OnSectionCompleted;
+        event SectionUnlocked OnSectionUnlocked;
+        event TaskProgressed OnTaskProgressed;
+
+        void InitializeQuests(string jsonMessage);
+        void InitializeQuests(List<QuestModel> parsedQuests);
+        void UpdateQuestProgress(string jsonMessage);
+        void UpdateQuestProgress(QuestModel progressedQuest);
+        void RemoveQuest(string jsonMessage);
+        void RemoveQuest(QuestModel quest);
+    }
+
+    public class QuestsController : MonoBehaviour, IQuestsController
```

Comment:
If this is used as bridge, I'd rename this class to `QuestsBridge` and move all the business logic to a simple C# class named `QuestsController` (think humble object pattern). If we pull off this, we can avoid the annoying MonoBehaviour lifecycle as best as possible and improve the legibility of what is the communication with kernel.

Reply:
Will do!